### PR TITLE
Add support for BaselineAdminNetworkPolicy.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Add support for sig-network-policy-api's BaselineAdminNetworkPolicy.

- Map the single BANP to a policy (with order=1000) in a new dedicated Tier with order=10,000,000.
- Convert the policy as per AdminNetworkPolicy.  Due to different datatypes, the logic has to be copied (I didn't fancy using reflection).
- Add RBAC to allow Typha and Node to access the resources.

## Related issues/PRs

CORE-10800

Builds on ANP work:

Tier support: https://github.com/projectcalico/calico/pull/9085
Default action in tiers: https://github.com/projectcalico/calico/pull/9232
AdminNetworkPolicy: #9206
AdminNetworkPolicy egress networks: https://github.com/projectcalico/calico/pull/9276
AdminNetworkPolicy named port: https://github.com/projectcalico/calico/pull/9254

## Todos

- [x] Tests
- [ ] Documentation
- [x] Release note

## Release Note

```release-note
Add support for Kubernetes BaselineAdminNetworkPolicy. https://network-policy-api.sigs.k8s.io/api-overview/
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: libcalico-go/lib/names/policy.go
Comment (mazdakn) [inline libcalico-go/lib/names/policy.go:30]: This prefix should be `kbanp.baselineadminnetworkpolicy`. This is long, but basically the second part `baselineadminnetworkpolicy` is the tier name.

Comment (mazdakn) [reply libcalico-go/lib/names/policy.go:30]: If we want to make it shorter, maybe we can name the tier `baselinepolicy` and this prefix `kbanp.baselinepolicy`.

Comment (fasaxc) [reply libcalico-go/lib/names/policy.go:30]: Good spot, thanks.

File: libcalico-go/lib/names/policy.go
Comment (mazdakn) [inline libcalico-go/lib/names/policy.go:29]: what is `ossg.`? I saw it before, and I meant to ask about it.

File: libcalico-go/lib/backend/k8s/conversion/conversion.go
Comment (mazdakn) [inline libcalico-go/lib/backend/k8s/conversion/conversion.go:581]: You probably forgot to replace this section with the new `unpackANPPorts` function.

**Review by mazdakn — APPROVED** [review]
LGTM. I just left a few nits.

File: libcalico-go/lib/backend/k8s/conversion/conversion.go
Comment (fasaxc) [reply libcalico-go/lib/backend/k8s/conversion/conversion.go:581]: This _is_ the new function?

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, squash the commits when I merge it, and leave the branch after I've merged it.

File: libcalico-go/lib/backend/k8s/conversion/conversion.go
Comment (mazdakn) [reply libcalico-go/lib/backend/k8s/conversion/conversion.go:581]: clearly I left this comment in a wrong place. I meant this change: https://github.com/projectcalico/calico/pull/9568

**Pull request merged**
